static/app/views/dashboards/widgetCard/autoSizedText.tsx

```diff
@@ -0,0 +1,143 @@
+import {useLayoutEffect, useRef} from 'react';
+import styled from '@emotion/styled';
+import * as Sentry from '@sentry/react';
+
+interface Props {
+  children: React.ReactNode;
+  maximumDifference?: number;
+}
+
+export function AutoSizedText({
+  children,
+  maximumDifference = DEFAULT_MAXIMUM_DIFFERENCE,
+}: Props) {
+  const childRef = useRef<HTMLDivElement>(null);
+
+  const fontSize = useRef<number>(0);
+  const fontSizeLowerBound = useRef<number>(0);
+  const fontSizeUpperBound = useRef<number>(0);
+
+  useLayoutEffect(() => {
+    const childElement = childRef.current; // This is `SizedChild`
+    const parentElement = childRef.current?.parentElement; // This is the parent of `AutoSizedText`
+
+    if (!childElement || !parentElement) {
+      return () => {};
+    }
+
+    // On component first mount, register a `ResizeObserver` on the containing element. The handler fires
+    // on component mount, and every time the element changes size after that
+    const observer = new ResizeObserver(entries => {
+      // The entries list contains an array of every observed item. Here it is only one element
+      const entry = entries[0];
+
+      if (!entry) {
+        return;
+      }
+
+      // The resize handler passes the parent's dimensions, so we don't have to get the bounding box
+      const parentDimensions = entry.contentRect;
+
+      // Reset the iteration parameters
+      fontSizeLowerBound.current = 0;
+      fontSizeUpperBound.current = parentDimensions.height;
+
+      let iterationCount = 0;
+
+      const span = Sentry.startInactiveSpan({
+        op: 'function',
+        name: 'AutoSizedText.iterate',
+      });
+
+      // Run the resize iteration in a loop. This blocks the main UI thread and prevents
+      // visible layout jitter. If this was done through a `ResizeObserver` or React State
+      // each step in the resize iteration would be visible to the user
+      while (iterationCount <= ITERATION_LIMIT) {
+        const childDimensions = getElementDimensions(childElement);
+
+        const widthDifference = parentDimensions.width - childDimensions.width;
+        const heightDifference = parentDimensions.height - childDimensions.height;
+
+        const childFitsIntoParent = heightDifference > 0 && widthDifference > 0;
+        const childIsWithinWidthTolerance =
+          Math.abs(widthDifference) <= maximumDifference;
+        const childIsWithinHeightTolerance =
+          Math.abs(heightDifference) <= maximumDifference;
+
+        if (
+          childFitsIntoParent &&
+          (childIsWithinWidthTolerance || childIsWithinHeightTolerance)
+        ) {
+          // Stop the iteration, we've found a fit!
+          break;
+        }
+
+        adjustFontSize(childDimensions, parentDimensions);
+
+        iterationCount += 1;
+      }
+
+      span.setAttribute('iterationCount', iterationCount);
```

Comment:
Wdyt about adding the final disparity as the attribute? This way, if you change the logic in the future, you can also evaluate how accurate the sizing algorithm is vs the iteration count
```suggestion
      span.setAttribute('disparity', final disparity);
```

Reply:
Makes sense! Added into the loop, alongside the iteration count